CloudM.Exercise.WebApi/Controllers/PackageController.cs

```diff
@@ -27,12 +31,63 @@ public async Task<PackageDependencies> Get(string name, string version)
                 PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
             });
 
+            var dependencies = npmPackage.Versions[version].Dependencies;
+
+            foreach (var dep in dependencies)
+            {
+                var subDep = GetDependencies(dep.Key, dep.Value).Result;
+                dependencyTree.Add(dep.Key, new
+                {
+                    Version = dep.Value,
+                    Dependecies = subDep
+                });
+            }
+
             return new PackageDependencies
             {
                 Name = name,
                 Version = version,
-                Dependencies = npmPackage.Versions[version].Dependencies,
+                Dependencies = dependencyTree,
             };
         }
+
+        public async Task<Dictionary<string, dynamic>> GetDependencies(string name, string version)
+        {
+            using var httpClient = new HttpClient();
+            var response = await httpClient.GetAsync($"https://registry.npmjs.org/{name}");
+            response.EnsureSuccessStatusCode();
+            var jsonString = await response.Content.ReadAsStringAsync();
+            var npmPackage = JsonSerializer.Deserialize<NpmPackage>(jsonString, new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            });
+
+            var range = new Range(version);
+            var v = range.MaxSatisfying(npmPackage.Versions.Keys);
+
+            if (!string.IsNullOrEmpty(v))
+            {
+                var dependencyTree = new Dictionary<string, dynamic>();
+                var newDeps = npmPackage.Versions[v].Dependencies;
+                if (newDeps != null)
+                {
+                    foreach (var dep in newDeps)
+                    {
+                        var subDep = GetDependencies(dep.Key, dep.Value).Result;
+                        dependencyTree.Add(dep.Key, new
+                        {
+                            Version = dep.Value,
+                            Dependecies = subDep
+                        });
+                    }
+                }
+
+                return dependencyTree;
+            }
+            else
+            {
+                return null;
```

Comment:
As per one of my upper comments - lets return the empty dictionary as opposed to Null